### DISCUSSION TITLE
How to query a tree relation

### DISCUSSION THREAD

**Discussion by TerranWorks** [SeaQL/sea-orm#1654] [Q&A]
I have two tables that look like this:
```sql
CREATE TABLE IF NOT EXISTS post_meta (
    id INT NOT NULL AUTO_INCREMENT,
    PRIMARY KEY (id),

    title VARCHAR(255) NOT NULL,
    created TIMESTAMP NOT NULL,
    updated TIMESTAMP NOT NULL
) ENGINE=INNODB;

CREATE TABLE IF NOT EXISTS post (
    id INT NOT NULL AUTO_INCREMENT,
    PRIMARY KEY (id),

    post_data INT NOT NULL,
    CONSTRAINT fk_post_data
    FOREIGN KEY (post_data)
    REFERENCES post_meta(id)
        ON UPDATE CASCADE
        ON DELETE CASCADE,

    original_post_data INT,
    CONSTRAINT fk_original_post_data
    FOREIGN KEY (original_post_data)
    REFERENCES post_meta(id)
        ON UPDATE CASCADE
        ON DELETE CASCADE
) ENGINE=INNODB;
```

I want to select the post and both of the post_data fields (apparently this is called a tree relation). I was referred to a potential solution in an [issue](https://github.com/SeaQL/sea-orm/issues/1637#issuecomment-1547386483) I posted while trying to figure this out. It seems like I want something like this:

```rust
    let posts: Vec<crate::PostModel> = crate::Post::find().all(db).await?;
    let post_metas: Vec<Vec<crate::PostMetaModel>> = posts.load_many(crate::PostMeta, db).await?;
```

But it requires Related to be implemented for Post however it doesn't look like it can be implemented twice so I can't see how I would specify two connections between the tables.

I'm sure its simple but I'm stuck right now. Any help is appreciated. I have tried to search for similar discussions but unfortunately the word "tree" is very common

**Comment by billy1624** [comment]
Hey @TerranWorks, `Related` can't be implemented multiple time but another API `Linked` can. https://www.sea-ql.org/SeaORM/docs/relation/chained-relations/
